app/screens/settings/display_settings/display_settings.js

```diff
@@ -32,9 +34,12 @@ export default class DisplaySettings extends PureComponent {
         showClockDisplaySettings: false,
     };
 
-    constructor(props) {
-        super(props);
-        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+    componentDidMount() {
+        this.navigationEventListener = Navigation.events().bindComponent(this);
+    }
+
+    componentDidAppear() {
+        setNavigatorStyles(this.props.componentId, this.props.theme);
```

Comment:
why are we setting the styles after the screen appeared? seems to be too late

Reply:
I'm fine with using `setNavigatorStyles` if the screen still transitions properly. The main reason I was leaning towards `Navigation.push` was that I was seeing some cases where the top bar colour or text wouldn't transition smoothly when going to certain screens or navigating back to them later on.